python_modules/libraries/dagster-k8s/dagster_k8s_tests/unit_tests/test_container_context.py

```diff
@@ -323,3 +360,35 @@ def test_merge(empty_container_context, container_context, other_container_conte
     assert container_context.merge(empty_container_context) == container_context
     assert empty_container_context.merge(container_context) == container_context
     assert other_container_context.merge(empty_container_context) == other_container_context
+
+    deep_merged_container_context = container_context.merge(
+        other_container_context._replace(
+            run_k8s_config=UserDefinedDagsterK8sConfig.from_dict(
+                {
+                    **other_container_context.run_k8s_config.to_dict(),
+                    "merge_behavior": K8sConfigMergeBehavior.DEEP.value,
+                }
+            )
+        ),
+    )
+
```

Comment:
would be nice to have a test case with obnoxiously deep merging to protect recursion